reconcile/gabi_authorized_users.py

```diff
@@ -0,0 +1,98 @@
+from datetime import datetime, date
+import logging
+import sys
+from typing import Dict, List, Optional
+
+import reconcile.queries as queries
+import reconcile.openshift_base as ob
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.openshift_resource import (OpenshiftResource as OR,
+                                                ResourceInventory as RI)
+from reconcile.utils.defer import defer
+
+QONTRACT_INTEGRATION = 'gabi-authorized-users'
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+EXPIRATION_MAX = 90
+
+
+def construct_gabi_oc_resource(name: str, users: List[str]) -> OR:
+    body = {
+        "apiVersion": "v1",
+        "kind": "ConfigMap",
+        "metadata": {
+            "name": name,
+            "annotations": {
+                "qontract.recycle": "true"
+            }
+        },
+        "data": {
+            "authorized-users.yaml": '\n'.join([u for u in users])
+        }
+    }
+    return OR(body, QONTRACT_INTEGRATION, QONTRACT_INTEGRATION_VERSION,
+              error_details=name)
+
+
+def fetch_desired_state(gabi_instances: List[Dict], ri: RI) -> None:
+    for g in gabi_instances:
+        exp_date = datetime.strptime(g['expirationDate'], '%Y-%m-%d').date()
+        if exp_date < date.today():
+            continue
+        elif (exp_date - date.today()).days > EXPIRATION_MAX:
+            logging.error(
+                f'The maximum expiration date of {g["name"]} '
+                f'shall not exceed {EXPIRATION_MAX} days form today')
+            sys.exit(1)
+        users = [u['github_username'] for u in g['users']]
+        resource = construct_gabi_oc_resource(g['name'], users)
+        for i in g['instances']:
+            namespace = i['namespace']
+            account = i['account']
+            identifier = i['identifier']
+            tf_resources = namespace['terraformResources']
+            found = False
+            for tf_resource in tf_resources:
+                if tf_resource['provider'] == 'rds' and \
+                        tf_resource['account'] == account and \
+                        tf_resource['identifier'] == identifier:
+                    found = True
+                    break
+            if not found:
+                logging.error(
+                    f'Could not find rds identifier {identifier} '
+                    f'for account {account} in namespace {namespace["name"]}')
+                sys.exit(1)
+            cluster = namespace['cluster']['name']
+            ri.add_desired(
+                cluster,
+                namespace['name'],
+                'ConfigMap',
+                g['name'],
+                resource
+            )
+
+
+@defer
+def run(dry_run: bool, thread_pool_size: int = 10,
+        internal: Optional[bool] = None, use_jump_host: bool = True,
```

Comment:
nit: Just to save you some typing in the future, `use_jump_host` doesn't need a type annotation, [it will be inferred](https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html) because of the default boolean value.